src/picosvg/svg_pathops.py

```diff
@@ -75,9 +75,18 @@ def _end_path(points) -> SVGCommandGen:
     "endPath": _end_path,
 }
 
+_SVG_FILL_RULE_TO_SKIA_FILL_TYPE = {
+    "nonzero": pathops.FillType.WINDING,
+    "evenodd": pathops.FillType.EVEN_ODD,
+}
+
 
-def skia_path(svg_cmds: SVGCommandSeq):
-    sk_path = pathops.Path()
+def skia_path(svg_cmds: SVGCommandSeq, fill_rule: str = "nonzero") -> pathops.Path:
```

Comment:
maybe don't default fill_rule here? - I think it's reasonable to insist clients specify

Reply:
If I don't default, then we'd have to also change `svg_pathops.transform`, `svg_pathops.stroke`, `svg_pathops.bounding_box` and `svg_pathops.union` and `interesection` as well, to pass on the fill_rule down to `skia_path`. But the first three do not really need/use fill_rule (the result won't change if path was interpreted as nonzero or evenodd); for the latter two (which use _do_pathop), they currently take Iterable[SVGCommandSeq] (with `*` args tuple unpacked); if we also need to pass the extra fill_rule, we need to either pass an additional list of fill_rules (one per SVGCommandSeq), or we change the input to be Iterable[Tuple[SVGCommandSeq, str]]. 

What I currently do is instead pass the fill_rule only when it matters (in remove_overlaps, where we evenodd/nonzero does make a difference). For union/intersection where it _would_ make a difference, I prevent that by first removing overlaps (thus fixing winding) of the shapes I will union/intersect (using clip_rule because I assume those will be used in a clipPath context).

Reply:
If the result of skia_path depends on the fill_rule then you should require it be passed explicitly. The complexity is fundamental to the operation and shouldn't be hidden. You know the correct value in the cited calling sites so specification there is correct.

Reply:
ok.. explicit is better than implicit. I was trying to avoid having to change many call sites.

> we need to either pass an additional list of fill_rules (one per SVGCommandSeq), or we change the input to be Iterable[Tuple[SVGCommandSeq, str]]

What do you suggest for `svg_pathops.union` and `svg_pathops.intersection`? How shall we pass down the fill_rule to skia_path?

Reply:
There is also the logical imbalance of requiring the fill_rule to make a skia_path, on the one hand, and on the other returning an SVGCommandGen which doesn't know/care about the SkPath.fillType, simply yields svg path commands (assuming "nonzero", because that is how we then go on to use those commands).

Reply:
in other words, we can't roundtrip an SVGPath through a pathops.Path and back to SVGPath without losing the original fill_rule or the Path.fillType as set by skia operations, because we are reducing the Skia Path to an iterable of SVG path commands.

Reply:
I guess that wherever we currently `return svg_commands(sk_path)` we should instead return a tuple that also includes the SVG fill_rules from the SkPaths' fillType:

```
return svg_commands(sk_path), svg_fill_rules(sk_path)`
```

Reply:
We can document the result, either by comment or by type. The minimal solution would be to specify in comments. For example intersection would be commented to returns command sequence based on a nonzero fill rule.

Better would be to have SVGCommandSeq have a fill_rule as one cannot reasonably interpret the commands without it. We could do that in this PR or we could finish this PR by adding comments and file an issue to have fill rule on command sequences in a future PR.

Reply:
after we perform boolean operations with skia-pathops the resulting paths are always FillType.WINDING, because we call them with the default `fix_winding=True`:
https://github.com/fonttools/skia-pathops/blob/6581155e5151624ae9756392db025810933b74d6/src/python/pathops/operations.py#L38

that's what we want indeed, we never want evenodd paths because OT fonts only support nonzero.
I added assertions to confirm that.

The only thing we currently don't support is converting a Skia Path with fillType.EVEN_ODD to SVG with fill_rule="evenodd", because `svg_commands` return value doesn't contain that information. As long the fill type doesn't change from "nonzero" to "evenodd" while inside Skia, this should be fine.

The results of transform, bounding_box or stroke operations is not affected by the original path's fill_rule property. So we can treat incoming SVG paths as "nonzero" and leave the fill rule of the resulting svg_commands unspecified in the case of transform, or assume it will be "nonzero" for stroked outlines.